### PR TITLE
♻️ `servicelib.fastapi` tools and `rabbitmq.rpc` errors interface

### PR DESCRIPTION
**Pull request by pcrespov**
#### ♻️  `fastapi` tools
Separation of policies so they can be combined at will (and not imposed from a base class)
- `app_state.AppStateMixin`: to create/get/delete a **single instance** of a class in the `app.state` 
- `http_client`: 
  - wrapper to `httpx.AsyncClient` to bind lifespan of the  wrapped client to the app
  - In a separate PR, we will split healthcheck mixin with liveness probes and readyness probes
- `httpx_utils`: free functions on `httpx` interface

#### ✨ RPC errors interface
Allows raising selection of exceptions in rpc (similar to https://tenacity.readthedocs.io/en/latest/#whether-to-retry)
   - *common error interface* : moves common `payments` exceptions `models_library.api_schemas_payments.errors.`
   - *rpc-server interface*: define in server side which exceptions are "expected":  set in `payments.api.rpc` interface: this way interface includes not only functionality but expected errors.
   - *plugin to http error*: now `payments` plugin will raise in the rpc calls the exceptions above and are handled accordingly in the rest calls. Note that the rest has yet another error handling mechanism that translates these exceptions to http errors. SEE `wallets._handlers.handle_wallets_exceptions`)
   - These are the diffrent errors
       - `_BaseAppError`: internal to `payments` service
       - `PaymentsPluginError`: internal to `payments` plugin in webserver
       - `_BaseRpcApiError` subclasses: shared by RPC server(in `payments` service) and client (in `web-server` service)

   
#### 🐛 fixes a bug that affect "FAKE" mode 
 - used for development i.e. inconvenient but not critical
 - adds more logs for fake mode

## Related issue/s

- This was part of coming https://github.com/ITISFoundation/osparc-simcore/pull/5125

## How to test

see new driving tests files

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

## DevOps

- Title added in .env-devel I will also add it in the `repo.config`s. This is a way to clarify which `env` I am using when I have symbolic links to them

### CODE REVIEW

File: packages/models-library/src/models_library/api_schemas_payments/errors.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+from pydantic.errors import PydanticErrorMixin
+
+
+class BaseServiceError(PydanticErrorMixin, ValueError):
```
Comment (sanderegg) [inline packages/models-library/src/models_library/api_schemas_payments/errors.py:4]: should be private?

File: packages/service-library/src/servicelib/rabbitmq/_rpc_router.py [hunk 2]
```diff
@@ -6,48 +6,73 @@
...
+                            exc_type=f"{exc.__class__.__module__}.{exc.__class__.__name__}",
                             msg=f"{exc}",
                         ) from None
 
-            self.routes[RPCMethodName(func.__name__)] = wrapper
+            self.routes[RPCMethodName(func.__name__)] = _wrapper
             return func
 
-        return decorator
+        return _decorator
```
Comment (sanderegg) [inline packages/service-library/src/servicelib/rabbitmq/_rpc_router.py:78]: great!

**Review by sanderegg — APPROVED** [review]
very nice! thanks!

File: packages/models-library/src/models_library/api_schemas_payments/errors.py (hunk 1 above)
Comment (pcrespov) [reply packages/models-library/src/models_library/api_schemas_payments/errors.py:4]: It's a good point. By making it private we are enforcing people to use more concrete exceptions in the interfaces which could be a good thing to do :-)

**Review by GitHK — APPROVED** [review]
look good to me

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks a lot!!

**Review by bisgaard-itis — APPROVED** [review]
Cool! Thanks a lot

**Pull request merged**